crates/terminal/src/terminal.rs

```diff
@@ -2033,6 +2045,27 @@ fn regex_match_at<T>(term: &Term<T>, point: AlacPoint, regex: &mut RegexSearch)
     visible_regex_match_iter(term, regex).find(|rm| rm.contains(&point))
 }
 
+// Check if the url ends with a valid delimiter and if the 
+// number of opening and closing delimiters is the 
+pub fn valid_url_ending(url: &str) -> bool {
```

Comment:
```suggestion
fn valid_url_ending(url: &str) -> bool {
```

* not a public function for sure
* overall, quite odd: regex is capable of saying "not ends with a particular item(s)" relatively easily and we have the `is_path_surrounded_by_common_symbols` for the exact check already

Why do even need this?

Reply:
Hi, so the reason why i did this was i had a hard time figuring out how to use regex for the case where we end with ')' or ']'.

this is better visualized with an example. For instance if we have 

(https://www.example.com)

the `regex_match_at` function will parse out the first '(' and return https://www.example.com) in this case we would just want https://www.example.com.

However, if we have the example, 

(https://www.example.com_(test))

then the `regex_match_at` function will parse out https://www.example.com_(test)) where we want https://www.example.com_(test)

I tried working with regex to so that i can include parentheses in the url for both cases but it kept so i introduced this function and conditional to overcome that.

Reply:
Yes, conditions are hard to check with regex, but cannot we use `is_path_surrounded_by_common_symbols` first, extract the "sanitized" substring and match that?